### PR TITLE
Modernize and update metadata for rustls fork

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

**Review by jsha — CHANGES_REQUESTED** [review]
In https://github.com/rustls/webpki/pull/1/commits/ef0c3d388dade898b21af7999e3386a6ae3ea8d2 you say Ubuntu 18 is deprecated. By whom? According to https://ubuntu.com/about/release-cycle it's supported by Ubuntu until April 2023. It's fine to drop support in this crate, but probably we should document that; or follow the example of \*ring\* and just document what platforms we test on rather than committing to supporting specific platforms.

**Comment by djc** [comment]
Sorry, yes, 18.04 is specifically deprecated for GitHub Actions:

https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,72 +6,45 @@ on:
   push:
 jobs:
   rustfmt:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
 
     steps:
-      - uses: briansmith/actions-rs-toolchain@v1
+      - uses: actions-rs/toolchain@v1
```
[the remark this answers is not in the document]
Comment (djc) [reply .github/workflows/ci.yml:12]: Maybe, but then we should also do that across the other repos in the rustls org? It feels like a big job and it's not obvious to me that there are big risks here.

So my take is that for now, Brian's forks are probably less- or equally well maintained compared to the originals (and there are currently no secrets in this repository, either).

**Review by jsha — APPROVED** [review]
Sounds good. 👍🏻

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctz) [reply .github/workflows/ci.yml:12]: Seems like there are two options for dealing with GH actions poor security defaults:

1. totally distrust the CI process: it gets read-only access to the repo (same as anything else on the internet), and no special access to secrets etc
2. carefully curate and control the CI process, so we can trust it with secrets and privileged repo access

I think we should be expending effort doing (1) -- and I've just checked this repo has minimal GITHUB_TOKEN permissions (which should be the case for the other rustls-org repos already)

**Pull request merged**